### PR TITLE
Fixes moths infinitely eating most clothing

### PR DESCRIPTION
**Pull request by ArcaneDefence**
## About The Pull Request
Creates a new `damage_flag` for the proc call done by moths eating clothing items, so that articles of clothing can be fully consumed, which is checked against melee armor.
Traitor objectives like the advanced magboots do not take integrity damage from this, and remain unaffected by this change.

This has been bugged since June 2020 as this change made it so that an article of clothing needed a bomb, fire, or acid damage flag to be completely destroyed. https://github.com/tgstation/tgstation/pull/50558#discussion_r697777901

## Why It's Good For The Game
Infinite sources of food available everywhere on your person is bad
Untitled

Moths eating the entirety of a piece of clothing is innately comical, and this is a much funnier outcome that can fuel social interactions with others
image

## Changelog
:cl:
fix: Moths have finally figured out how to finish eating clothes again, rather than infinitely eating a singular sock.
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
world hunger created

**Comment by maxymax13** [comment]
> world hunger created

noooooooooo

**Comment by AMonkeyThatCodes** [comment]
@Mothblocks just remove it I beg you

**Comment by ATH1909** [comment]
Will this make bomb suits take longer (or possibly forever?) to fully chew, or does take_damage() not work that way?

**Comment by ArcaneDefence** [comment]
> 
> 
> Will this make bomb suits take longer (or possibly forever?) to fully chew, or does take_damage() not work that way?

It does, but if that's considered a problem I'll just add a vore damage flag and everyone will be worse off for it

**Comment by Mothblocks** [comment]
i would rather a separate damage flag just because you have to admit this is a hack

File: code/__DEFINES/combat.dm [hunk 1]
```diff
@@ -41,6 +41,8 @@
 #define MAGIC "magic"
 /// Involved in checking the likelyhood of applying a wound to a mob.
 #define WOUND "wound"
+/// Involves being consumed
+#define VORE "vore"
```
Comment (Mothblocks) [inline code/__DEFINES/combat.dm:45]: dont name it VORE name it something like moth_eating dummy

Comment (Mothblocks) [reply code/__DEFINES/combat.dm:45]: or consume

Comment (ArcaneDefence) [reply code/__DEFINES/combat.dm:45]: I have a mouth and I must feed

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -492,13 +492,20 @@ BLIND     // can't see anything
 		//so the shred survives potential turf change from the explosion.
 		addtimer(CALLBACK_NEW(/obj/effect/decal/cleanable/shreds, list(T, name)), 1)
 		deconstruct(FALSE)
+	if(damage_flag == CONSUME) //This allows for moths to fully consume clothing, rather than damaging it like other sources like brute
+		var/turf/current_position = get_turf(src)
+		new /obj/effect/decal/cleanable/shreds(current_position, name)
+		if(isliving(loc))
+			var/mob/living/possessing_mob = loc
+			possessing_mob.visible_message(span_danger("[src] is consumed until naught but shreds remains!"), span_warning("<b>[src] falls apart into little bits!</b>"))
```
Comment (Mothblocks) [inline code/modules/clothing/clothing.dm:500]:
```suggestion
			possessing_mob.visible_message(span_danger("[src] is consumed until naught but shreds remains!"), span_boldwarning("[src] falls apart into little bits!"))
```

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**